contracts/yield-source/YearnV2YieldSource.sol

```diff
@@ -191,7 +191,7 @@ contract YearnV2YieldSource is IYieldSource, ERC20Upgradeable, OwnableUpgradeabl
         }
         uint256 currentBalance = token.balanceOf(address(this));
 
-        return previousBalance.sub(currentBalance);
+        return currentBalance > previousBalance ? currentBalance.sub(previousBalance) : 0;
```

Comment:
I'm not sure the vulnerability report is accurate. 
I think it would make more sense if `previousBalance` was named `balanceBefore` and `currentBalance`,  `balanceAfter`.
Before withdrawing from the vault, `token.balanceOf(address(this));` should be higher than after withdrawing, so to me it makes sense to write `beforeBalance.sub(afterBalance);`
Also `vault.withdraw(yShares);` on line `190` should be `vault.withdraw(yShares, address(this));`, vault shares need to be minted to the yield source contract and not the sender.
https://yearn.github.io/yearn-devdocs/v2/smart-contracts/vault#withdraw-2

Reply:
Agreed....I don't think the check is needed.  It might be needed if there were rounding errors, but it doesn't appear possible as it's just a token balance.